### DISCUSSION TITLE
How to give different values to datasources through different environments？

### DISCUSSION THREAD

**Discussion by ilgnefz** [prisma/prisma#18294] [Q&A] [answered]
I have solved the problem of multiple `.dev` files by using `dotenv-cli`, but I still have some questions.

My two environment files are `.env.development` and `.env.production`
```
DATABASE_URL="mysql://root:0000@localhost:3306/dev"
```
```
DATABASE_URL="mysql://root:0000@localhost:3306/pro"
```
 The following modified data sources are used according to the document:
```typescript
import { Injectable } from '@nestjs/common';
import { PrismaClient } from '@prisma/client';

@Injectable()
export class PrismaService extends PrismaClient {
  constructor() {
    super({
      datasources: {
        db: {
          url: `${process.env.DATABASE_URL}`,
        },
      },
    });
  }
}
```
In `main.ts`, I used `dotenv` to change the env file:
```typescript
import { ValidationPipe } from '@nestjs/common';
import { NestFactory } from '@nestjs/core';
import { AppModule } from './app.module';
import { config } from 'dotenv';

async function bootstrap() {
  const app = await NestFactory.create(AppModule);
  app.useGlobalPipes(new ValidationPipe());
  const envFile =
    process.env.NODE_ENV === 'production'
      ? '.env.production'
      : '.env.development';
  config({ path: envFile });
  console.log(process.env.DATABASE_URL);
  await app.listen(3000);
}
bootstrap();
```
When I run `npx Prisma migrate dev`, the terminal will report an error:
```
Error: Prisma schema validation - (get-config wasm)
Error code: P1012
error: Environment variable not found: DATABASE_URL.
  -->  schema.prisma:10
 9 |   provider = "mysql"
10 |   url      = env("DATABASE_URL")
   |

Validation Error Count: 1
```
Because I changed the name of the `.dev` file to `.env.development`.
If I want to modify `db` through `datasource`, what should I do?

**Comment by janpio** [comment, marked as answer]
You get this error because when you run `npx prisma migrat edev` no other code is executed before that could load the `.env.development` file. You can use [`dotenv-cli` ](https://www.npmjs.com/package/dotenv-cli)for that and run that before you do `npx prisma migrate dev` so the environment variables from your file are already loaded.

**Reply by ilgnefz** [reply]
It turns out that the two documents `using-multiple-env-files` and `Programmatically override a datasource ur` should be used together. Thank you for your answer.
